edb/edgeql/compiler/stmt.py

```diff
@@ -779,9 +779,12 @@ def compile_Shape(
             f'shapes cannot be applied to '
             f'{expr_stype.get_verbosename(ctx.env.schema)}'
         )
-    view_type = viewgen.process_view(
-        stype=expr_stype, path_id=expr.path_id,
-        elements=shape.elements, ctx=ctx)
+    try:
+        view_type = viewgen.process_view(
+            stype=expr_stype, path_id=expr.path_id,
+            elements=shape.elements, ctx=ctx)
+    except irast.InvalidScopeConfiguration as e:
```

Comment:
`InvalidScopeConfigration` gets thrown by `ScopeNode.attach_path()`, so perhaps it'll be better to invent a wrapper around it that takes source context as argument and reraises?

Reply:
Should we just drop InvalidScopeConfiguration (or at least drop the node arguments and add source context), and pass in source contexts on all the call sites? There don't seem to be that many.

Reply:
The original motivation behind a bespoke exception was to keep `scopetree` entirely self-contained and unit-testable, and the intended purpose of the node arguments was to handle it and make the error message a bit more helpful by finding the source context for the other appearance of the symbol.  Alas, those aspirations didn't really materialize yet.  Let's drop the node arguments.